src/chainparamsbase.cpp

```diff
@@ -48,4 +48,5 @@ std::unique_ptr<CBaseChainParams> CreateBaseChainParams(const std::string& chain
 void SelectBaseParams(const std::string& chain)
 {
     globalChainBaseParams = CreateBaseChainParams(chain);
+    gArgs.SelectConfigSection(chain);
```

Comment:
I'm not sure if it makes sense to call this from `SelectBaseParams()`. This function is about setting the blockchain parameters, not the global node config.

Doesn't it make sense just to set `gArgs.m_section` directly at the end of `ReadConfigFile()`?

Reply:
I don't think so -- `ReadConfigFile()` only gets to know `-testnet` was set, you need to call `GetChainName()` to work out that means `chain=="test"`, and the place that's currently called is when passing the arguments to `SelectParams` or `SelectBaseParams`. Also, `SelectBaseParams` is already updating `globalChainBaseParams` which seems like "global node config" to me. :)

Changing that might not be a bad idea (and I think it might benefit jtimon's #8994), but I think it could wait for a different PR?

Reply:
fine. Can wait for a future PR.